go.mod

```diff
@@ -6,87 +6,86 @@ require (
 	github.com/Masterminds/semver v1.5.0
 	github.com/alecthomas/kong v0.2.17
 	github.com/crossplane/crossplane-runtime v0.16.0
-	github.com/google/go-cmp v0.5.7
+	github.com/google/go-cmp v0.5.8
 	// TODO(hasheddan): we prefer to consume release versions of
 	// go-containerregistry. An incremental version is currently being used to
 	// consume the new k8schain implementation, which fixes an issue with
 	// reconcile timeouts when pulling packages in some environments.
-	github.com/google/go-containerregistry v0.8.1-0.20220302183023-329563766ce8
-	github.com/google/go-containerregistry/pkg/authn/k8schain v0.0.0-20220302183023-329563766ce8
+	github.com/google/go-containerregistry v0.8.1-0.20220414143355-892d7a808387
```

Comment:
@smileisak thanks so much for this update! It looks like the fix was included in https://github.com/google/go-containerregistry/pull/1346, which was in the most recent release (https://github.com/google/go-containerregistry/releases/tag/v0.9.0). Could we pin to `v0.9.0` and drop my `TODO` comment above?

Reply:
Yes, I though that we still have the issue of the reconcile timeouts when pulling packages in some environments. I'll update this MR to use the `v0.9.0`.

Reply:
@smileisak the update where we moved from a release version (e.g. `v0.7.0`) to an incremental version (e.g. `v0.8.1-0.20220302183023-329563766ce8`) was to fix the problem described. Because there wasn't a release version available we had to use the incremental version even though we prefer not to. Now that the previous fix, and the one you are including in this PR are in a release version `v0.9.0` we can depend on it and drop my `TODO` as it is no longer applicable :)